### PR TITLE
Add question check before matching & show confirmation popup before leaving matching pages

### PR DESCRIPTION
**Pull request by jolynloh**

### CODE REVIEW

File: frontend/src/pages/Home/index.tsx [hunk 1]
```diff
@@ -42,10 +49,23 @@ const Home: React.FC = () => {
   }
   const { findMatch, loading } = match;
 
+  const isSmallerThan1100px = useMediaQuery('(max-width:1100px)');
+
   useEffect(() => {
     getQuestionCategories(dispatch);
   }, []);
 
+  useEffect(() => {
```
Comment (feliciagan) [inline frontend/src/pages/Home/index.tsx:58]: how come need to listen to state.questions ahh? is it possible to just include all the functionality here into the button onclick function?

Comment (jolynloh) [reply frontend/src/pages/Home/index.tsx:58]: ohh cuz if i immediately check state.questions after calling getQuestionList it doesn't work at times i think the state sometimes doesn't update immediately?

Comment (ruiqi7) [reply frontend/src/pages/Home/index.tsx:58]: yeah, I think because dispatch is like fire-and-forget so the state might not update in time

**Review by ruiqi7 — APPROVED** [review]

File: frontend/src/pages/Home/index.tsx (hunk 1 above)
Comment (feliciagan) [reply frontend/src/pages/Home/index.tsx:58]: ohh ok

**Pull request merged**